### PR TITLE
update LOAD DATA ... REPLACE compatibility

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

With https://github.com/pingcap/tidb/pull/31303 merged we now support `LOAD DATA...REPLACE`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- Other reference link(s): ttps://github.com/pingcap/tidb/pull/31303

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-load-data.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ In the above example, `x'2c'` is the hexadecimal representation of the `,` chara
 
 ## MySQL compatibility
 
-This statement is understood to be fully compatible with MySQL except for the `LOAD DATA...REPLACE INTO` syntax [#24515](https://github.com/pingcap/tidb/issues/24515). Any other compatibility differences should be [reported via an issue](https://github.com/pingcap/tidb/issues/new/choose) on GitHub.
+This statement is understood to be fully compatible with MySQL. Any other compatibility differences should be [reported via an issue](https://github.com/pingcap/tidb/issues/new/choose) on GitHub.
```
Comment (morgo) [inline sql-statements/sql-statement-load-data.md:108]:
```suggestion
This statement is understood to be fully compatible with MySQL. Any compatibility differences should be [reported via an issue](https://github.com/pingcap/tidb/issues/new/choose) on GitHub.
```

**Comment by dveeden** [comment]
@morgo PTAL

**Comment by TomShawn** [comment]
@dveeden Does it apply to v5.4 or v6.0?

**Comment by dveeden** [comment]
> @dveeden Does it apply to v5.4 or v6.0?

I think it applies to v6.0, but not to v5.4.

**Review by morgo — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by morgo** [comment]
@dveeden Sorry, some additional changes :( In this text:

> In addition, TiDB currently only supports parsing the syntax of the DuplicateOpt, CharsetOpt, and LoadDataSetSpecOpt parameters.

The `DuplicateOpt` refers to REPLACE compatibility - so that text is out of date. `LoadDataSetSpecOpt` also looks like it is implemented, but `CharsetOpt` is not.

It is easier to read when we put all the compatibility issues under "MySQL compatibility" and not burried in the text. What do you think about removing this line completely - and changing the compatibility statement to say only utf8mb4 charset is supported for importing (other charsets are ignored)?

**Pull request merged**